third_party/efr32_sdk/efr32_firmware_utils.py

```diff
@@ -0,0 +1,317 @@
+
+        The generated script is a minimal wrapper around `flash_command()`,
+        containing any option values that differ from the class defaults.
+        """
+
+        # Note: this modifies the argument parser, so the same Flasher instance
+        # should not be used for both parse_argv() and make_wrapper().
+        self.parser.description = 'Generate a flashing script.'
+        self.parser.add_argument(
+            '--output',
+            metavar='FILENAME',
+            required=True,
+            help='flashing script name')
+        self.parser.add_argument(
+            '--scripts-dir',
+            metavar='DIR',
+            required=True,
+            help='script utilities directory')
+        args = self.parser.parse_args(argv)
+
+        # Find any option values that differ from the class defaults.
+        # These will be inserted into the wrapper script.
+        defaults = []
+        for key, value in vars(args).items():
+            key = key.replace('_', '-')
+            if key in self.options and value != self.options[key]:
+                defaults.append('  {}: {},'.format(repr(key), repr(value)))
+
+        script = """
+            import sys
+
+            SCRIPTS_DIR = '{scripts_dir}'
+            DEFAULTS = {{
+            {defaults}
+            }}
+
+            sys.path.append(SCRIPTS_DIR)
+            import {module}
+
+            if __name__ == '__main__':
+                sys.exit({module}.flash_command(sys.argv[1:], DEFAULTS))
+        """
+
+        script = ('#!/usr/bin/env python' + textwrap.dedent(script).format(
+            scripts_dir=args.scripts_dir,
+            module=__name__,
+            defaults='\n'.join(defaults)))
+
+        try:
+            with open(args.output, 'w') as script_file:
+                script_file.write(script)
+            os.chmod(args.output, (stat.S_IXUSR | stat.S_IRUSR | stat.S_IWUSR
+                                   | stat.S_IXGRP | stat.S_IRGRP
+                                   | stat.S_IXOTH | stat.S_IROTH))
+        except OSError as exception:
+            print(exception, sys.stderr)
+            return 1
+        return 0
+
+
+class EFR32Flasher(Flasher):
+    """Manage efr32 flashing."""
+
+    def __init__(self, options=None):
+        super().__init__(options)
+        self.define_options(EFR32_OPTIONS)
+
+    def commander(self, arguments):
+        """Run commander."""
+        command = [self.options['commander']]
+        command += arguments
+        self.log(3, 'Execute:', *command)
+        self.err = subprocess.call(command)
+        return self
+
+    def commander_logging(self,
+                          arguments,
+                          name,
+                          pass_message=None,
+                          fail_message=None,
+                          fail_level=0):
+        """Run commander with log messages."""
+        self.log(1, name)
+        if self.commander(arguments).err:
```

Comment:
Can we usefully detect the "commander not in PATH" case here and issue an error message to that effect, maybe with a pointer to the README or the download location for commander?

Reply:
Good idea; I'll look in to how to check for that. (Python's `subprocess` doesn't appear to distinguish that from other failures.)

Reply:
Just to be clear, this PR should not block on that.  Followup is just fine.